### ISSUE TITLE
CPU usage peak when triggering Hotque's on 5.33 ms setting on i7 quadcore os x Mojave

### ISSUE THREAD

**Issue by Tommy-Rutting** [mixxxdj/mixxx#13101]
### Bug Description

When i trigger a hotque on opposite deck it creates a CPU spike on it's own deck and is audible on the other deck.
I'm using Rubberband R3, when both tracks are playing CPU is at 40%, when using hotque or moving the platter then there is major CPU spiking.
When doing it quick it extremly drops the tempo/processing on the other deck.
Running Traktor and or Serato DJ with keylock and same latency runs perfectly on same machine Quadcore 2.3 Ghz i7 on Mojave. I also have Quadcore i7 2,6 Ghz on Catalina that is even worse. Renice did not solve it etc.
Problem is on internal audio and Traktor Audio 6 interface.
Is R3 Rubberband usable, Rubberband is to low-fi?

### Version

2.4

### OS

macOS Mojave 10.14.5 Quadcore i7 2.3 Ghz

**Comment by ronso0** [comment]
Yes, R3 is demanding. You may check if this is a general issue or if you can avoid it with increased latency.

> and is audible on the other deck.

buffer underruns mean Mixxx couldn't process the Main mix (or parts of it) in time, so they'll always be audible no matter the origin.

**Comment by Tommy-Rutting** [comment]
21.3 ms is the max i can get out of R3, that's unusable for live scratching and quick mixing. I like Mixxx (respect for the developers!), but i need around 5 ms latency for live performances.

I'm professional DJ using Serato Dj from 2005 on and want to switch to Mixxx.

Hope you will get it optimized/fixed.

**Comment by ronso0** [comment]
You don't like how R2 sounds?

**Comment by Tommy-Rutting** [comment]
Mmmm, sounds much flatter/compressed, but is workable.

**Comment by daschuer** [comment]
When seeking through the track, we fade the new buffer into the old buffer. This is to avoid clicks, but it requires also a momentary double CPU time. I your case we hunt one issue with another. 

We may discuss other solutions, like fading to silence and than fade into the new buffer form silence.

**Comment by Tommy-Rutting** [comment]
Thanks for explaining the chosen process, it sounds like a non usable solution for R3 and probably the other algoritms that are less CPU intensive but not noticable. Just saying maybe it's a stupid thought, can't you do something with zero crossing editing the buffer.

**Comment by daschuer** [comment]
Without crossing we need to fade out and in to avoid clicks. This will be notable as a small loudness drop.
We may mix the old buffer back to back which will introduce a click sound. 

A solution that does the crossfader on the unstreched audio would help, but this will introduce a notable delay, because the Rubber and buffers are already filled with the sound stream without the jump. 

Another issue is the way underflows are handled. If it happens the old buffer is played twice, and then the missed buffer is processed. It would be much better if the missed buffer is shipped. This will avoid that the crowd also talks out of rhythm.
